### PR TITLE
Keep support for PHPCR-ODM 1.4

### PR DESCRIPTION
**Pull request by dbu**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

as discussed on slack, lets keep support for phpcr-odm 1.4 so we can release without needing phpcr-odm 2.0 first.

### CODE REVIEW

File: src/Doctrine/Phpcr/RedirectRoute.php [hunk 1]
```diff
@@ -95,6 +95,24 @@ public function getParentDocument()
     }
 
     /**
+     * @deprecated For BC with the PHPCR-ODM 1.4 HierarchyInterface
+     * @see setParentDocument
+     */
+    public function setParent($parent)
+    {
+        return $this->setParentDocument($parent);
```
Comment (wouterj) [inline src/Doctrine/Phpcr/RedirectRoute.php]: Please also add deprecation triggers:

```php
@trigger_error('The '.__METHOD__.'() method is deprecated and will be removed in version 3.0. Use setParentDocument() instead.', E_USER_DEPRECATED);
```

Comment (dbu) [inline src/Doctrine/Phpcr/RedirectRoute.php]: right. just pushed them.

File: src/Doctrine/Phpcr/Route.php [hunk 2]
```diff
@@ -105,6 +105,28 @@ public function getParentDocument()
...
+        return $this->setParentDocument($parent);
+    }
+
+    /**
+     * @deprecated For BC with the PHPCR-ODM 1.4 HierarchyInterface
+     * @see getParentDocument
+     */
+    public function getParent()
+    {
+        @trigger_error('The '.__METHOD__.'() method is deprecated and will be removed in version 3.0. Use setParentDocument() instead.', E_USER_DEPRECATED);
```
Comment (wouterj) [inline src/Doctrine/Phpcr/Route.php]: should be `getParentDocument()` here (and in the other class)

Comment (dbu) [inline src/Doctrine/Phpcr/Route.php]: hah. how to copy paste...

**Comment by wouterj** [comment]
Pushed a fix for the failing build. As Symfony 3.0 is no longer maintained, it has some bugs with latest dependencies, causing our tests to fail. We should simply stop testing against this dep.

**Pull request merged**